gluten-hudi/src/main/scala/org/apache/gluten/execution/HudiScanTransformerProvider.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gluten.execution
+
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.execution.FileSourceScanExec
+
+class HudiScanTransformerProvider extends DataSourceScanTransformerRegister {
+
+  override val scanClassName: String = "org.apache.hudi.HoodieFileIndex"
```

Comment:
Should we change the `scanClassNam` from `file index` to `file format` according to https://github.com/apache/incubator-gluten/pull/3955/files#diff-7b2924f5d19292df8ae434c562c54924ea90de8a7be38669570cf7cb4d13402dR47

Reply:
it's actually better to stick to what the trait specifies for the variable name. See docs of `org.apache.gluten.execution.DataSourceScanTransformerRegister#scanClassName`
 
>    * For DataSource V1, it should be the child class name of
>   * [[org.apache.spark.sql.execution.datasources.FileIndex]].
>   *
>   * For DataSource V2, it should be the child class name of
>   * [[org.apache.spark.sql.connector.read.Scan]].